test/integration/client-side-encryption/client_side_encryption.prose.12.deadlock.test.ts

```diff
@@ -30,20 +31,26 @@ const $jsonSchema = BSON.EJSON.parse(
   )
 );
 
-class CapturingMongoClient extends MongoClient {
-  commandStartedEvents: Array<CommandStartedEvent> = [];
-  clientsCreated = 0;
-  constructor(url: string, options: MongoClientOptions = {}) {
-    options = { ...options, monitorCommands: true, __skipPingOnConnect: true };
-    if (process.env.MONGODB_API_VERSION) {
-      options.serverApi = process.env.MONGODB_API_VERSION as MongoClientOptions['serverApi'];
-    }
+function makeClient(
```

Comment:
The capturing mongoclient causes issues because clients are not instantiated with configuration.newClient().

Alternatively, I could make the TestConfiguration take a generic argument that extends MongoClient and have the tests provide the client constructor to use for new clients, but I chose instead to wrap the configuration's method.